rmrk-token/src/children.rs

```diff
@@ -0,0 +1,361 @@
+                    .entry(parent_token_id)
+                    .and_modify(|c| {
+                        c.insert(child_vec.clone());
+                    })
+                    .or_insert_with(|| {
+                        let mut a = BTreeSet::new();
+                        a.insert(child_vec.clone());
+                        a
+                    });
+            } else {
+                panic!("RMRKCore: child does not exist or has already been accepted");
+            }
+        }
+        self.children_status
+            .insert(child_vec, ChildStatus::Accepted);
+        msg::reply(
+            RMRKEvent::AcceptedChild {
+                child_token_address: *child_contract_id,
+                child_token_id,
+                parent_token_id,
+            },
+            0,
+        )
+        .unwrap();
+    }
+
+    /// Rejects an RMRK child being in the `Pending` status
+    /// Requirements:
+    /// * The `msg::source()` must be an RMRK owner or an approved account
+    /// Arguments:
+    /// * `parent_token_id`: is the tokenId of the parent NFT
+    /// * `child_contract_id`: is the address of the child RMRK contract
+    /// * `child_token_id`: is the tokenId of the child instance
+    pub async fn reject_child(
+        &mut self,
+        parent_token_id: TokenId,
+        child_contract_id: &ActorId,
+        child_token_id: TokenId,
+    ) {
+        let root_owner = self.find_root_owner(parent_token_id).await;
+        self.assert_approved_or_owner(parent_token_id, &root_owner);
+
+        // get the vector of `child_nft_contract` + `child_token_id`
+        let child_vec = get_child_vec(child_contract_id, child_token_id);
+
+        if let Some(children) = self.pending_children.get_mut(&parent_token_id) {
+            if children.contains(&child_vec) {
+                children.remove(&child_vec);
+                // sends message to child contract to burn RMRK token from it
+                burn_from_parent(child_contract_id, vec![child_token_id], &root_owner).await;
+            } else {
+                panic!("RMRKCore: child does not exist or has already been approved");
+            }
+        } else {
+            panic!("RMRKCore: there are no pending children");
+        }
+        self.children_status.remove(&child_vec);
+        msg::reply(
+            RMRKEvent::RejectedChild {
+                child_token_address: *child_contract_id,
+                child_token_id,
+                parent_token_id,
+            },
+            0,
+        )
+        .unwrap();
+    }
+
+    /// Rmoves an RMRK child being in the `Accepted` status
+    /// Requirements:
+    /// * The `msg::source()` must be an RMRK owner or an approved account
+    /// Arguments:
+    /// * `parent_token_id`: is the tokenId of the parent NFT
+    /// * `child_contract_id`: is the address of the child RMRK contract
+    /// * `child_token_id`: is the tokenId of the child instance
+    pub async fn remove_child(
```

Comment:
`remove_child` is very similar to `reject_child` part of the logic can be united